### PR TITLE
fix(lab): Debug and fix some typos

### PR DESCRIPTION
**Pull request by nchilada**
I think this PR should fix some occasional† failures that I was encountering in the v3 draft view.

† I'm not sure why the failures were so unpredictable; maybe because of stale data cached in localStorage?

### CODE REVIEW

File: sites/lab/package.json [hunk 1]
```diff
@@ -4,8 +4,10 @@
   "private": true,
   "scripts": {
     "dev": "node --experimental-json-modules ../../node_modules/.bin/next dev -p 8000",
+    "dev-debug": "node --inspect --experimental-json-modules ../../node_modules/.bin/next dev -p 8000",
```
Comment (nchilada) [inline sites/lab/package.json:7]: There's got to be a better way to inspect the lab, one that doesn't require a whole new command…unfortunately I was having problems with `NODE_OPTIONS='--inspect' yarn lab`.

**Comment by BenJamesBen** [comment]
The `develop` localhost lab has been having lots of weird state problems and/or timing issues. I'll test this PR tomorrow to see what issues it clears up for me. 

FYI, to try to avoid problems, my best practice currently is to:
1. Restart yarn lab.
2. Clear cache and cookies, and load the fresh localhost lab.
3. Go to a design and immediately ClearThing.
4. Select a measurement and go to DesignDraft view.
5. Immediately reload the web page in the browser.

**Comment by nchilada** [comment]
> The `develop` localhost lab has been having lots of weird state problems and/or timing issues. I'll test this PR tomorrow to see what issues it clears up for me.
> 
> FYI, to try to avoid problems, my best practice currently is to:
> 
>     1. Restart yarn lab.
> 
>     2. Clear cache and cookies, and load the fresh localhost lab.
> 
>     3. Go to a design and immediately ClearThing.
> 
>     4. Select a measurement and go to DesignDraft view.
> 
>     5. Immediately reload the web page in the browser.
> 
> 
> Edit: added: I was able to test the PR changes just now. The PR localhost lab seems to work just fine for me, though it did not fix any of the state issues I've been seeing. That is, it seems to work as well as the current `develop` branch version.

@BenJamesBen interesting, I'll try to keep those pointers in mind! And thanks anyway for checking this PR.

File: packages/core/src/utils.mjs [hunk 2]
```diff
@@ -682,13 +689,21 @@ const __addPartOptionalMeasurements = (part, config, store, list = false) => {
 const __addPartOptions = (part, config, store) => {
   if (part.options) {
     for (const optionName in part.options) {
-      if (!config.optionDistance[optionName]) {
+      if (!config.optionDistance) {
+        config.optionDistance = {}
+      }
+      if (!config.optionDistance.hasOwn(optionName)) {
```
Comment (BenJamesBen) [inline packages/core/src/utils.mjs:695]: This line seems to be causing an error in my localhost lab:
```
Unhandled Runtime Error
TypeError: config.optionDistance.hasOwn is not a function
```

**Comment by joostdecock** [comment]
Hi everyone,
There was some weird issues in core that were caused by a lack of immutability of the design parts and options.
That's been resolved in the work I pushed today. I think it would be good to rebase this PR and see what is still relevant after those changes.

thanks and sorry for the trouble

**Comment by nchilada** [comment]
@joostdecock sure, I'll rebase and see what still applies...but FWIW, I would not be offended if you prefer to incorporate these observations into your active work instead of merging this PR. I can't keep up with your rate of work on the `develop` branch!

@BenJamesBen good catch and sorry for the late reply! I'll fix that if it's still relevant after rebasing.

**Comment by joostdecock** [comment]
> I would not be offended if you prefer to incorporate these observations into your active work instead of merging this PR. I can't keep up with your rate of work on the develop branch!

@nchilada I've put in the lab-debug changes, the jsdoc improvements and some other things in https://github.com/freesewing/freesewing/commit/e960d31d96c6c71d74103b150adbe4be9c9fb513
Sorry about not being able to tend to this sooner, things are a bit hectic right now :shrug: 

Let me know if you want to keep this open an rebase (again) or close it.

**Comment by nchilada** [comment]
@joostdecock nice! And I'm sorry for not updating this, I ended up being busier than I'd expected this week. If there's anything left in this PR, I can take a look at it this weekend, otherwise you can certainly feel free to close it.

**Comment by joostdecock** [comment]
I feel this is now rather stale. Closing.

**Pull request closed without merging**